File: app/src/main/java/com/huanchengfly/tieba/post/components/dividers/HorizontalSpacesDecoration.java
package com.huanchengfly.tieba.post.components.dividers;

import android.graphics.Rect;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class HorizontalSpacesDecoration extends RecyclerView.ItemDecoration {
    private int top;
    private int bottom;
    private int left;
    private int right;

    public HorizontalSpacesDecoration(int space) {
        this(space, space, space, space);
    }

    public HorizontalSpacesDecoration(int top, int bottom, int left, int right) {
        this.top = top;
        this.bottom = bottom;
        this.left = left;
        this.right = right;
    }

    @Override
    public void getItemOffsets(@NonNull Rect outRect, @NonNull View view,
                               @NonNull RecyclerView parent, @NonNull RecyclerView.State state) {
        int position = parent.getChildAdapterPosition(view);
        RecyclerView.Adapter adapter = parent.getAdapter();
        if (adapter == null) {
            return;
        }
        if (position == 0) {
            outRect.left = left;
            outRect.right = right / 2;
        } else if (position == adapter.getItemCount() - 1) {
            outRect.left = left / 2;
            outRect.right = right;
        } else {
            outRect.left = left / 2;
            outRect.right = right / 2;
        }
        outRect.bottom = bottom;
        outRect.top = top;
    }
}

File: app/src/main/java/com/huanchengfly/tieba/post/components/dividers/HorizontalSpacesDecoration.kt
package com.huanchengfly.tieba.post.components.dividers

import android.graphics.Rect
import android.view.View
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ItemDecoration

class HorizontalSpacesDecoration @JvmOverloads constructor(
    private val top: Int,
    private val bottom: Int,
    private val left: Int,
    private val right: Int,
    private val spaceOnEdge: Boolean = true
) : ItemDecoration() {
    constructor(space: Int) : this(space, space, space, space)

    constructor(space: Int, spaceOnEdge: Boolean) : this(space, space, space, space, spaceOnEdge)

    override fun getItemOffsets(
        outRect: Rect, view: View,
        parent: RecyclerView, state: RecyclerView.State
    ) {
        val position = parent.getChildAdapterPosition(view)
        val adapter = parent.adapter ?: return
        if (position == 0) {
            outRect.left = if (spaceOnEdge) left else 0
            outRect.right = right / 2
        } else if (position == adapter.itemCount - 1) {
            outRect.left = left / 2
            outRect.right = if (spaceOnEdge) right else 0
        } else {
            outRect.left = left / 2
            outRect.right = right / 2
        }
        outRect.bottom = bottom
        outRect.top = top
    }
}